Add GitLab crosswalk

Aidajafarbigloo opened the pull request:
This pull request adds a crosswalk between CodeMeta and GitLab.

@dgarijo 
Could you please review the additions?

Aidajafarbigloo committed:
Add GitLab crosswalk

progval commented on crosswalks/GitLab.csv:
```diff
@@ -0,0 +1,74 @@
+targetProduct,
+applicationCategory,
+applicationSubCategory,
+downloadUrl,http_url_to_repo
```
```suggestion
downloadUrl,
```

https://schema.org/downloadUrl says "If the file can be downloaded, URL to download the binary." which does not match `http_url_to_repo`.

Aidajafarbigloo replied:
Was removed in Commit 0517459.

progval commented on crosswalks/GitLab.csv:
```diff
@@ -0,0 +1,74 @@
+roleName,
+startDate,/projects/:id/members/ -> created_at
+softwareSuggestions,
+maintainer,projects/:id/members/all -> access level:40
```
I'm not sure about the `->` here. Other mappings use a "flat" mapping. I'll leave this to @dgarijo.

Aidajafarbigloo replied:
all `->` was removed.

dgarijo commented:
Thanks for the quick review! We will build on this

dgarijo commented on crosswalks/GitLab.csv:
```diff
@@ -0,0 +1,74 @@
+Property,GitLab
+codeRepository,web_url
```
in GitHub crosswalk, we have `codeRepository` as `html_url`. Here, it looks like web url matches the same intent. We should be consistent with the mappings, so the group believed web_url was ok here.

In the example, it says: `"http_url_to_repo": "http://example.com/diaspora/diaspora-project-site.git"`, while `"web_url": "http://example.com/diaspora/diaspora-project-site"`. 

In sssom mappings, both are mapped as correct.
Reviewed only until here

Aidajafarbigloo commented:
The crosswalk targets [GitLab API v4](https://docs.gitlab.com/api/rest/#make-a-rest-api-request), as of October 2025.

dgarijo commented on crosswalks/GitLab.csv:
```diff
@@ -0,0 +1,74 @@
+releaseNotes,
+softwareHelp,
+softwareRequirements,
+softwareVersion,/projects/:id/releases
```
That is true. The name or `tag_name` in the releases seem more appropriate. But this would give you the mapping for all releases, you would have to select the latest release, which is the first element (e.g., https://gitlab.com/api/v4/projects/18483189/releases).
So it would be something like: `/projects/:id/releases[0]:tag_name`, right?

Aidajafarbigloo replied:
Was updated in Commit f46c3a0.

Aidajafarbigloo committed:
Remove 'http_url_to_repo'

In response to https://github.com/codemeta/codemeta/pull/444#discussion_r2472500276 
and https://github.com/codemeta/codemeta/pull/444#discussion_r2982749523

Aidajafarbigloo committed:
Update softwareVersion property match

In response to https://github.com/codemeta/codemeta/pull/444#discussion_r2472503143, https://github.com/codemeta/codemeta/pull/444#discussion_r2627931317, https://github.com/codemeta/codemeta/pull/444#discussion_r2817087564, https://github.com/codemeta/codemeta/pull/444#discussion_r2817091307.

Aidajafarbigloo committed:
Fix URL field 

Keeping only `http_url_to_repo `. In response to https://github.com/codemeta/codemeta/pull/444#discussion_r2472508115 and https://github.com/codemeta/codemeta/pull/444#discussion_r2982765325.

Aidajafarbigloo commented on crosswalks/GitLab.csv:
```diff
@@ -0,0 +1,74 @@
+identifier,id
+name,name
+sameAs,
+url,http_url_to_repo OR web_url
```
Was updated in Commit 0f5330d (mapped to `http_url_to_repo`). Please let me know if it needs to be changed.

Aidajafarbigloo commented on crosswalks/GitLab.csv:
```diff
@@ -0,0 +1,74 @@
+applicationSubCategory,
+downloadUrl,http_url_to_repo
+fileSize,/projects/:id/repository/files/:file_path  (File Info)
+installUrl,projects/:id/releases/:tag_name/assets/links
```
/projects/:id/releases/:tag_name → assets.sources[format=zip].url

dgarijo replied:
To keep it consistent with GitHub API mapping, this is mapping is (for the moment) discarded.

Aidajafarbigloo replied:
Was removed in Commit 4255a93.

Aidajafarbigloo commented on crosswalks/GitLab.csv:
```diff
@@ -0,0 +1,74 @@
+name,name
+sameAs,
+url,http_url_to_repo OR web_url
+relatedLink,wiki_enabled OR _links
```
`_links` in `/projects/:id/releases/:tag_name` contains:
- closed_issues_url
- closed_merge_requests_url
- merged_merge_requests_url
- opened_issues_url
- opened_merge_requests_url
- self

Aidajafarbigloo replied:
`_links` removed in Commit 5f27e88.

Aidajafarbigloo committed:
Fix relatedLink field

Removed `_links` in response to https://github.com/codemeta/codemeta/pull/444#discussion_r2982797817.

Aidajafarbigloo committed:
Remove proposed match to `fileSize`

In response to https://github.com/codemeta/codemeta/pull/444#discussion_r2982908048.

Aidajafarbigloo commented on crosswalks/GitLab.csv:
```diff
@@ -0,0 +1,74 @@
+applicationCategory,
+applicationSubCategory,
+downloadUrl,http_url_to_repo
+fileSize,/projects/:id/repository/files/:file_path  (File Info)
```
Was removed in Commit e8ecbc8.

Aidajafarbigloo committed:
Update contributor field 

In response to https://github.com/codemeta/codemeta/pull/444#discussion_r2982959311.

Aidajafarbigloo commented on crosswalks/GitLab.csv:
```diff
@@ -0,0 +1,74 @@
+supportingData,
+author,creator_id
+citation,
+contributor,/projects/:id/members OR _links.members
```
Updated in Commit ca8fd13.

Aidajafarbigloo replied:
`/projects/:id/members/all` shows inherited members as well.

Aidajafarbigloo committed:
Update dateModified field 

In response to https://github.com/codemeta/codemeta/pull/444#discussion_r2982963563.

Aidajafarbigloo commented on crosswalks/GitLab.csv:
```diff
@@ -0,0 +1,74 @@
+copyrightHolder,
+copyrightYear,
+dateCreated,created_at
+dateModified,last_activity_at OR updated_at
```
Updated in Commit 1dc5438.

Aidajafarbigloo commented on crosswalks/GitLab.csv:
```diff
@@ -0,0 +1,74 @@
+datePublished,
+editor,
+encoding,
+fileFormat,/projects/:id/repository/files/:file_path  (File Info)
```
`/projects/:id/repository/files/:file_path  (File Info)` is not publicly available. 

/projects/:id/releases/:tag_name → assets.sources.format

dgarijo replied:
Since this applies to the format of a distribution of a release, it may not make a lot of sense to apply it globally to the whole repo. Suggestion in today's meeting is to drop this mapping (also for consistency in GitHub API crosswalk)

Aidajafarbigloo replied:
Was removed in Commit fb6a40b.

Aidajafarbigloo committed:
Switch OR to /

In response to https://github.com/codemeta/codemeta/pull/444#discussion_r2982986190.

Aidajafarbigloo commented on crosswalks/GitLab.csv:
```diff
@@ -0,0 +1,74 @@
+encoding,
+fileFormat,/projects/:id/repository/files/:file_path  (File Info)
+funder,
+keywords,topics OR tag_list
```
Updated in Commit b262226.

Aidajafarbigloo committed:
Update fileds

In response to https://github.com/codemeta/codemeta/pull/444#discussion_r2983002536.

Aidajafarbigloo commented on crosswalks/GitLab.csv:
```diff
@@ -0,0 +1,74 @@
+sameAs,
+url,http_url_to_repo OR web_url
+relatedLink,wiki_enabled OR _links
+givenName,/projects/:id/members/ -> name
```
Some updated in Commit b1b4473.

dgarijo commented on crosswalks/GitLab.csv:
```diff
@@ -0,0 +1,74 @@
+roleName,
+startDate,/projects/:id/members/all/:id/created_at
+softwareSuggestions,
+maintainer,projects/:id/members/all/:id/access level:40
```
I would say this type of mapping is not supported in this version of the crosswalks, as it implies "if the access level of a member is 40, then it is a maintainer".

dgarijo replied:
This is mapping the level of the members to maintainer, which is not correct. I suggest to drop this mapping.

Aidajafarbigloo replied:
Was removed in Commit 906ede1.

dgarijo commented on crosswalks/GitLab.csv:
```diff
@@ -0,0 +1,74 @@
+Property,GitLab
+codeRepository,web_url
+programmingLanguage,/projects/:id/languages
```
This comment can be resolved (for some reason I can't resolve it)

Aidajafarbigloo committed:
Fix fileFormat 

In response to https://github.com/codemeta/codemeta/pull/444/changes/BASE..0afec5e0d1e05ec7291ad2b7fd07c920f9a362f0#r3046133972

Aidajafarbigloo committed:
Remove the proposed match to maintainer

In response to https://github.com/codemeta/codemeta/pull/444#discussion_r3046162345

Aidajafarbigloo committed:
Remove the proposed match to installUrl

In response to https://github.com/codemeta/codemeta/pull/444#discussion_r3046112040

dgarijo reviewed (approved):
@moranegg please merge this PR (on behalf of the working group on Codemeta, April 21st meeting)
